aspect/build_defs.bzl

```diff
@@ -11,22 +19,63 @@ _java_8_transition = transition(
     outputs = ["//command_line_option:javacopt"],
 )
 
-def _java_8_cfg_impl(ctx):
-    files = []
+def _aspect_files_impl(ctx):
+    files = depset(
+        transitive =
+            [it[DefaultInfo].files for it in ctx.attr.files] +
+            [it[DefaultInfo].files for it in ctx.attr.jars],
+    )
 
-    for jar in ctx.attr.jars:
-        files.extend(jar[DefaultInfo].files.to_list())
+    return [
+        DefaultInfo(files = files),
+        AspectFilesInfo(files = files, namespace = ctx.attr.namespace),
```

Comment:
do we need to store `files` in this provider? is there a reason we can't use `DefaultInfo`?